loom/common/core/allocator.h

```diff
@@ -76,6 +76,63 @@
  *
  */
 
+#define LOOM_ALLOCATOR_CHECK 1
```

Comment:
This whole complex would benefit from a big comment block indicating purpose and theory of operation. Probably two paragraphs minimum as there's a lot going on here.